zilliqa/src/api/eth.rs

```diff
@@ -292,10 +292,24 @@ fn get_transaction_count(params: Params, node: &Arc<Mutex<Node>>) -> Result<Stri
     expect_end_of_params(&mut params, 3, 3)?;
 
     let node = node.lock().unwrap();
+
     let block = node.get_block(block_id)?;
     let block = build_errored_response_for_missing_block(block_id, block)?;
 
-    Ok(node.get_state(&block)?.get_account(address)?.nonce.to_hex())
+    let nonce = node.get_state(&block)?.get_account(address)?.nonce;
+
+    if matches!(block_id, BlockId::Number(BlockNumberOrTag::Pending)) {
+        let pending_transaction_count = node
+            .txpool_content()
+            .pending
```

Comment:
Annoyingly I think we need to be slightly smarter with this algorithm. If the pool contains many transactions from a signer, but they aren't contiguous with the signer's current nonce, then we shouldn't increment the nonce.

i.e.
account.nonce = 0
pool = [{ nonce = 1 }, { nonce = 2 }]
The current implementation would return `transaction_count = 2` when we actually want `transaction_count = 0`